driver: remove `mktemp` tool dependency in `ShellDriver`

flxzt opened the pull request:
**Description**

This removes the use of the `mktemp` tool on targets when transferring files through the `ShellDriver`.
It is replaced by `touch /tmp/tmp-<random-uuid>` which should be always available since `touch` is part of the GNU core utilities.

**Checklist**
- [x] PR has been tested

jluebbe commented:
This is not equivalent to the old behavior, as touch may create files readable by other users. Also, touch doesn't fail if the file already exists.

At least, this change needs a good reasoning why this changes is an improvement and doesn't break existing usecases.

flxzt commented:
I can only speak for my use-case but it is the following: I came across a target that has only the most basic tools available, not much more than coreutils and does not have the ability to easily add additional tools when testing the production image.
I had hoped this dependency removal is something which benefits other users with similar targets as well.
I can't see how it breaks existing use-cases since the created tempfile really is only used for a short while until it's contents are `dd`'ed to the actual destination.

I didn't know that file permissions are a concern here, let me add a chmod 600 which should replicate the `mktemp`'s default behaviour (unless you have another suggestion).

Regarding not failing when a file with the same name already exists - If you want a check can be added before calling `touch`, but the possibility of collision of a UUID4 which the to-be-created file contains in it's path according to wikipedia : `number of random version-4 UUIDs which need to be generated in order to have a 50% probability of at least one collision is 2.71 quintillion` [source](https://en.wikipedia.org/wiki/Universally_unique_identifier#Collisions).

sjg20 commented on labgrid/driver/shelldriver.py:
```diff
@@ -364,10 +365,8 @@ def _target_cleanup(tmpfile):
-            raise ExecutionError('Could not make temporary file on target')
-        tmpfile = tmpfile[0]
+        tmpfile = f'/tmp/tmp-{uuid4()}'
+        self._run_check(f'touch {tmpfile}')
```
How about only doing this as a workaround when mktemp fails? Then we can keep using the normal approach for most cases?

flxzt replied:
That would work for me - as a fallback. What are your thoughts about it @jluebbe ?

flxzt committed:
driver: Fallible `mktemp` command in ShellDriver

This changes that the creation of a temporary file
falls back to using touch if the `mktemp` tool fails,
which happens when it is not available on the target.

Signed-off-by: Felix Zwettler <Felix.Zwettler@duagon.com>
